### PR TITLE
Search is back

### PR DESCRIPTION
**Pull request by grischa**
Switching to haystack 2.4 and an elasticsearch backend by default, search is now working again for simple title/name searches.

full metadata search is also going to come, but this is useful enough to roll out first.

### CODE REVIEW

File: tardis/search/views.py [hunk 1]
```diff
@@ -290,14 +259,55 @@ def create_response(self):
         return render_response_index(self.request, self.template, context)
 
 
-@login_required
-def single_search(request):
-    search_query = FacetFixedSearchQuery(backend=HighlightSearchBackend())
-    sqs = SearchQuerySet(query=search_query)
-    sqs.highlight()
+# @login_required
+# def single_search(request):
```
Comment (jasonrig) [inline tardis/search/views.py]: Function commented out and seems unused; probably should remove.

File: tardis/default_settings.py [hunk 2]
```diff
@@ -377,14 +378,26 @@ def get_admin_media_path():
 
 # Settings for the single search box
 # Set HAYSTACK_SOLR_URL to the location of the SOLR server instance
-SINGLE_SEARCH_ENABLED = False
-HAYSTACK_SITECONF = 'tardis.search_sites'
-HAYSTACK_SEARCH_ENGINE = 'solr'
-HAYSTACK_SOLR_URL = 'http://127.0.0.1:8080/solr'
+SINGLE_SEARCH_ENABLED = True
```
Comment (jasonrig) [inline tardis/default_settings.py]: I just noticed that search is enabled by default. Probably not a good idea, @grischa.

**Comment by grischa** [comment]
This is done now, @jasonrig

**Comment by jasonrig** [comment]
Could you please rebase onto develop? Github's complaining that the branch is out of date.

**Comment by grischa** [comment]
Ah, sorry about that. Done

**Comment by pansapiens** [comment]
It looks like the `'tardis.search'` app and routing to _/search_ is active even if the search box is turned off (eg I can still attempt to search via _/search?q=somequery_ ). Should the `'tardis.search'` app be removed from INSTALLED_APPS and the _/search/_ routing disabled in urls.py when `SINGLE_SEARCH_ENABLED = False` ? Unless `'tardis.search'` is required for some other purpose when the UI element is turned off, it might be worth disabling to as to prevent exploratory 'mining' of the database via search on sites that have intentionally turned it off.

**Pull request merged**